Source/EventFlow/ReadStores/ReadStoreManager.cs

```diff
@@ -66,6 +66,23 @@ static ReadStoreManager()
             }
 
             AggregateEventTypes = new HashSet<Type>(iAmReadModelForInterfaceTypes.Select(i => i.GetTypeInfo().GetGenericArguments()[2]));
+            if (AggregateEventTypes.Count != iAmReadModelForInterfaceTypes.Count)
+            {
+                throw new ArgumentException(
+                    $"Read model type '{StaticReadModelType.PrettyPrint()}' implements ambiguous '{typeof(IAmReadModelFor<,,>).PrettyPrint()}' interfaces");
+            }
+        }
+
+        private static bool IsReadModelFor(Type i)
+        {
+            if (!i.GetTypeInfo().IsGenericType)
```

Comment:
Maybe save the type info and check for `IsInterface` to save the call to `GetGenericTypeDefinition` in a few cases

Reply:
It is only used in this context:

```csharp
    .GetInterfaces()
    .Where(IsReadModelFor)
```